src/eventSchemas/providers/schemasDataProvider.ts

```diff
@@ -0,0 +1,105 @@
+ */
+export class SchemasDataProvider {
+    private static INSTANCE: SchemasDataProvider | undefined
+    private readonly logger: Logger = getLogger()
+
+    public constructor(private readonly credentials: Credentials, private readonly cache: Cache) {}
+
+    public async getRegistries(region: string, client: SchemaClient) {
+        const cachedRegion = this.cache.regionDataList.filter(x => x.region === region).shift()
+        try {
+            // if region is not cached, make api query and retain results
+            if (!cachedRegion || cachedRegion.registryNames.length === 0) {
+                const registrySummary = await toArrayAsync(client.listRegistries())
+                const registryNames = registrySummary.map(x => x.RegistryName!)
+                this.pushRegionDataIntoCache(region, registryNames, [])
+
+                return registryNames
+            }
+        } catch (err) {
+            const error = err as Error
+            this.logger.error('Error retrieving registries', error)
+
+            return undefined
+        }
+
+        return cachedRegion!.registryNames
+    }
+
+    public async getSchemas(region: string, registryName: string, client: SchemaClient) {
+        const registrySchemasMapList = this.cache.regionDataList.filter(x => x.region === region).shift()
+            ?.registrySchemasMapList
+        let schemas = registrySchemasMapList?.filter(x => x.registryName === registryName).shift()?.schemaList
+        try {
+            // if no schemas found, make api query and retain results given that registryName && region already cached
+            if (!schemas || schemas.length === 0) {
+                schemas = await toArrayAsync(client.listSchemas(registryName))
+                const singleItem: registrySchemasMap = { registryName: registryName, schemaList: schemas }
+                //wizard setup always calls getRegistries method prior to getSchemas, so this shouldn't be undefined
+                if (!registrySchemasMapList) {
+                    this.pushRegionDataIntoCache(region, [], [singleItem])
+                }
+
+                if (registrySchemasMapList) {
+                    registrySchemasMapList.push(singleItem)
+                }
+            }
+        } catch (err) {
+            const error = err as Error
+            this.logger.error('Error retrieving schemas', error)
+
+            return undefined
+        }
+
+        return schemas
+    }
+
+    private pushRegionDataIntoCache(
+        region: string,
+        registryNames: string[],
+        registrySchemasMapList: registrySchemasMap[]
+    ): void {
+        const regionData: regionRegistryMap = {
+            region: region,
+            registryNames: registryNames,
+            registrySchemasMapList: registrySchemasMapList
+        }
+
+        this.cache.regionDataList.push(regionData)
+    }
+
+    public static getInstance(credential: Credentials, cache: Cache = new Cache([])): SchemasDataProvider {
```

Comment:
This is causing side effects on the singleton object, replacing it when there are different credentials.

getInstance shouldn't accept any parameters. 
  * If credential is used as a key cache, it becomes a concern of the cache. It would be passed to getXXX calls as needed.
  * You'd pass `new Cache([])` to the constructor on line 100. When you want to test SchemasDataProvider, you instantiate SchemasDataProvider objects, and provide a custom cache then, instead of calling getInstance

Reply:
Definitely, will update the code and unit tests as well